sdk/src/trace/batch_span_processor.cc

```diff
@@ -97,23 +99,23 @@ bool BatchSpanProcessor::ForceFlush(std::chrono::microseconds timeout) noexcept
   // Fix timeout to meet requirement of wait_for
   timeout = opentelemetry::common::DurationUtil::AdjustWaitForTimeout(
       timeout, std::chrono::microseconds::zero());
-  bool result;
-  if (timeout <= std::chrono::microseconds::zero())
+  std::chrono::steady_clock::duration timeout_steady =
+      std::chrono::duration_cast<std::chrono::steady_clock::duration>(timeout);
+  if (timeout_steady <= std::chrono::steady_clock::duration::zero())
   {
-    bool wait_result = false;
-    while (!wait_result)
-    {
-      // When is_force_flush_notified.store(true) and force_flush_cv.notify_all() is called
-      // between is_force_flush_pending.load() and force_flush_cv.wait(). We must not wait
-      // for ever
-      wait_result = synchronization_data_->force_flush_cv.wait_for(lk_cv, schedule_delay_millis_,
-                                                                   break_condition);
-    }
-    result = true;
+    timeout_steady = std::chrono::steady_clock::duration::max();
```

Comment:
I am not able to understand this part of code. The `AdjustWaitForTimeout` call above will convert `duration::zero()` to `duration::max()` and this line now will convert it back to `duration::zero()`.

Reply:
`AdjustWaitForTimeout` just cahnge the timeout to meet the requirement of `wait_for` .It will return the second parameter when the timeout is greater than max available value.It will change timeout to `duration::zero()` here, not to `duration::max()`.
Some old codes use zero for indefinite waiting. Maybe we can raise another PR to remove this implementation if we have cleanup all zero value usage. Maybe it is easier to understand to do not wait when we get zero here.
What's your suggestion?

Reply:
Yes, I think cleaning up the zero-value usage for indefinite waiting in a separate PR would be good. Thanks for the explanation.